### PR TITLE
Forum improvements

### LINKED ISSUE #868: Implement voting in the forum

### PR DESCRIPTION
**Pull request by lowjoel**
- Fix calculation of topic views
 - Implementing ordering topics by date
 - Show user profile pictures in topics
 - Show the vote counts in the topic listing

Fixes #868.

### CODE REVIEW

File: app/models/course/forum/topic.rb [hunk 1]
```diff
@@ -13,6 +13,16 @@ class Course::Forum::Topic < ActiveRecord::Base
...
   belongs_to :forum, inverse_of: :topics
 
+  # @!attribute [r] vote_count
+  #   The number of votes in this topic.
+  calculated :vote_count, (lambda do
+    Course::Discussion::Post::Vote.joins { post.topic }.
+      where do
+        (course_forum_topics.id == course_discussion_topics.actable_id) &
+          (course_discussion_topics.actable_type == Course::Forum::Topic.name)
+      end.select { count('*') }
```
Comment (lowjoel) [inline app/models/course/forum/topic.rb]: @allenwq running Rubocop locally is fine :confused:

**Comment by lowjoel** [comment]
@allenwq the Hakiri warning is a false positive, I think.

**Comment by allenwq** [comment]
@lowjoel Can fix the hakiri warning ? otherwise the subsequent PRs will all fail ?

**Comment by lowjoel** [comment]
That will be done only when the PR is merged.

File: app/models/course/forum/topic.rb [hunk 2]
```diff
@@ -31,6 +41,13 @@ class Course::Forum::Topic < ActiveRecord::Base
     end.select { count('*') }
   end)
 
+  # @!method self.order_by_date
+  #   Orders the topics by their date. This uses the modification date, so topics with new posts
```
Comment (allenwq) [inline app/models/course/forum/topic.rb]: by their dates ?  :-)

Comment (lowjoel) [inline app/models/course/forum/topic.rb]: :blush:

**Pull request merged**